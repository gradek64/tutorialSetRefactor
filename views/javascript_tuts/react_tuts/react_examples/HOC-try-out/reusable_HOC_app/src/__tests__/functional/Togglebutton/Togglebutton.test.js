import React from 'react';
// full render needed
import { mount } from 'enzyme';

/*
  *@this next line is defined in a root/jest.configuration.js
  *@so no need to import library
*/
// import toJson from 'enzyme-to-json';

// component for snapshot test;
import ToggleButton from '../../../functional/Togglebutton';

let wrapper;
beforeAll(() => {
  wrapper = mount(<ToggleButton
    action={() => {}}
    label="sort"
    toggleLabels={['ASC', 'DESC']}
    values={['sortASC', 'sortDESC']}
  />);
});

test('renders correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('It has SimpleButton component rendered', () => {
  expect(wrapper.find('SimpleButton').length).toBe(1);
});

test('it changes SimpleButton label once clicked', () => {
  // [ 'ASC', 'DESC' ]
  const labelsProps = wrapper.find('ToggleSimpleButton').prop('toggleLabels');

  // button clicked
  const SimpleButton = wrapper.find('SimpleButton');
  SimpleButton.find('button').simulate('click');
  // changes to ASC
  const expectedLabel = `sort ${labelsProps[1]}`;
  // expect label change
  expect(SimpleButton.prop('label')).toBe(expectedLabel);
});

test('it changes SimpleButton label back on once clicked', () => {
  // [ 'ASC', 'DESC' ]
  const labelsProps = wrapper.find('ToggleSimpleButton').prop('toggleLabels');

  // button clicked
  const SimpleButton = wrapper.find('SimpleButton');
  SimpleButton.find('button').simulate('click');
  // changes to ASC
  const expectedLabel = `sort ${labelsProps[0]}`;
  // expect label change
  expect(SimpleButton.prop('label')).toBe(expectedLabel);
});
