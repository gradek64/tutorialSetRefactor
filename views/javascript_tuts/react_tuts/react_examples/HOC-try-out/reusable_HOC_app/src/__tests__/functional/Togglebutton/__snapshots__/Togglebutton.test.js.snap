// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ToggleSimpleButton
  action={[Function]}
  label="sort"
  toggleLabels={
    Array [
      "ASC",
      "DESC",
    ]
  }
  values={
    Array [
      "sortASC",
      "sortDESC",
    ]
  }
>
  <ToggleState
    render={[Function]}
  >
    <SimpleButton
      eventHandler={[Function]}
      label="sort DESC"
      value=""
    >
      <button
        className="simple-button"
        onClick={[Function]}
      >
        sort DESC
      </button>
    </SimpleButton>
  </ToggleState>
</ToggleSimpleButton>
`;
